responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -205,6 +209,64 @@ def get_test_data(self, test_data=None):
                         columns=self._feature_metadata.dropped_features,
                         axis=1)
 
+    def _consolidate_categorical_features(
+            self,
+            categorical_features: Optional[List[str]],
+            feature_metadata: Optional[FeatureMetadata]):
+        """Consolidates the categorical features.
+
+        Originally, only the RAIInsights constructor accepted
+        categorical_features. Later on, feature_metadata was added as an
+        argument that also includes categorical_features.
+        This method consolidates them or raises an exception if it is not
+        possible. The resulting categorical features should be set on the
+        self._feature_metadata object. Eventually, the categorical_features
+        argument on the RAIInsights constructor will be removed at which
+        point this method can be removed as well.
+
+        :param categorical_features: the categorical features from the
+            RAIInsights constructor
+        :type categorical_features: Optional[List[str]]
+        :param feature_metadata: the feature metadata specified which may
+            include information on categorical features
+        :type feature_metadata: Optional[FeatureMetadata]
+        """
+        consolidated_categorical_features = []
+        if categorical_features is not None:
+            warnings.warn("The categorical_features argument on the "
+                          "RAIInsights constructor is deprecated and will "
+                          "be removed after version 0.26. Please provide "
+                          "categorical features via the feature_metadata "
+                          "argument instead.")
+        if feature_metadata is None:
+            # initialize to avoid having to keep checking if it is None
+            feature_metadata = FeatureMetadata()
+        if feature_metadata.categorical_features is None:
+            if categorical_features is None:
+                consolidated_categorical_features = []
+            else:
+                consolidated_categorical_features = categorical_features
+        else:
+            if categorical_features is None:
```

Comment:
should we check None OR empty list?  Like user might have passed an empty list for the param, but that probably just means we should ignore it.

Reply:
Great question! The way I see it there are a few cases involving empty lists: (I'll omit the flipped case)

- `(None, [])` -> go with second one, so no check for empty list needed
- `([], [])` -> neither is None and they match! That's another case where the sets are being compared.
- `([], ['a', ...])` -> neither is None and the set comparison doesn't match, raise exception.

If the user passes one as empty list explicitly and the other with actual categorical feature names, then that should raise an exception and make them aware of the mismatch. At least that's how I was looking at this. It's a super simple fix, too. I'll mark this as resolved, but do lmk if you feel like this isn't right.